apps/framework-cli-e2e/test/utils/database-utils.ts

```diff
@@ -0,0 +1,164 @@
+import { createClient } from "@clickhouse/client";
+import { ChildProcess } from "child_process";
+import { CLICKHOUSE_CONFIG, RETRY_CONFIG } from "../constants";
+import { withRetries } from "./retry-utils";
+
+/**
+ * Cleans up ClickHouse data by truncating test tables
+ */
+export const cleanupClickhouseData = async (): Promise<void> => {
+  console.log("Cleaning up ClickHouse data...");
+  await withRetries(
+    async () => {
+      const client = createClient(CLICKHOUSE_CONFIG);
+      try {
+        const result = await client.query({
+          query: "SHOW TABLES",
+          format: "JSONEachRow",
+        });
+        const tables: any[] = await result.json();
+        console.log(
+          "Existing tables:",
+          tables.map((t) => t.name),
+        );
+
+        await client.command({ query: "TRUNCATE TABLE IF EXISTS Bar" });
+        console.log("Truncated Bar table");
+
+        const mvTables = ["BarAggregated", "bar_aggregated"];
+        for (const table of mvTables) {
+          try {
+            await client.command({
+              query: `TRUNCATE TABLE IF EXISTS ${table}`,
+            });
+            console.log(`Truncated ${table} table`);
+          } catch (error) {
+            console.log(`Failed to truncate ${table}:`, error);
+          }
+        }
+      } finally {
+        await client.close();
+      }
+    },
+    {
+      attempts: RETRY_CONFIG.DEFAULT_ATTEMPTS,
+      delayMs: RETRY_CONFIG.DEFAULT_DELAY_MS,
+    },
+  );
+  console.log("ClickHouse data cleanup completed successfully");
+};
+
+/**
+ * Waits for database write operations to complete
+ */
+export const waitForDBWrite = async (
+  _devProcess: ChildProcess,
+  tableName: string,
+  expectedRecords: number,
+  timeout: number = 60_000,
+): Promise<void> => {
+  const attempts = Math.ceil(timeout / 1000); // Convert timeout to attempts (1 second per attempt)
+  await withRetries(
+    async () => {
+      const client = createClient(CLICKHOUSE_CONFIG);
+      try {
+        const result = await client.query({
+          query: `SELECT COUNT(*) as count FROM ${tableName}`,
+          format: "JSONEachRow",
+        });
+        const rows: any[] = await result.json();
+        const count = parseInt(rows[0].count);
+        console.log(`Records in ${tableName}:`, count);
+        if (count >= expectedRecords) {
```

Comment:
Should we be strict with this check? I'm thinking the test should know how many rows it generated, so if it's less or more then something is sus